low-level/framework/messaging/ingress_processor.py

```diff
@@ -185,31 +185,44 @@ def _process_msg(self, body):
                     "IngressProcessor, Authentication failed on message: %s" % ingressMsg)
                 return
 
+            # Check for debugging being activated in the message header
+            self._check_debug(message)
+            self._log_debug("_process_msg, ingressMsg: %s" % ingressMsg)
+
             # Get the incoming message type
             if message.get("actuator_request_type") is not None:
                 msgType = message.get("actuator_request_type")
 
                 # Validate against the actuator schema
                 validate(ingressMsg, self._actuator_schema)
+                # Compare hostname from the request to determine
+                # if request is meant for the current node
+                target_node_id = message.get("target_node_id")
+                if target_node_id is None:
+                    logger.warning(
+                        "Required attribute target_node_id is missing "
+                        "from actuator request")
+                    return
+                elif target_node_id == self._node_id:
+                    self._send_to_msg_handler(msgType, message, uuid)
+                else:
+                    logger.info(
+                        "Node identifier mismatch, actuator request rejected.")
+                    return
 
             elif message.get("sensor_request_type") is not None:
                 msgType = message.get("sensor_request_type")
 
                 # Validate against the sensor schema
                 validate(ingressMsg, self._sensor_schema)
+                self._send_to_msg_handler(msgType, message, uuid)
```

Comment:
why this change? relating to sensor rt !

Reply:
I have just rearranged the code. Earlier there were no checks so there was just one call of `_send_to_msg_handler()` for both sensor and actuator requests. Now since we have `target_node_id` for actuator, I added separate call for sensor requests.